code/components/extra-natives-five/src/RopeNatives.cpp

```diff
@@ -0,0 +1,146 @@
+#include "StdInc.h"
+
+#include "ropeManager.h"
+
+#include <Hooking.h>
+#include <ScriptEngine.h>
+#include <ScriptSerialization.h>
+
+namespace rage
+{
+static ropeDataManager* g_ropeDataManager;
+
+ropeDataManager* ropeDataManager::Get()
+{
+	return g_ropeDataManager;
+}
+
+static hook::thiscall_stub<rope*(ropeManager* manager, uint32_t handle)> ropeManager_getRopeFromHandle([]()
+{
+	return hook::get_call(hook::get_pattern("E8 ? ? ? ? 4C 8B E8 48 89 45 67"));
+});
+
+rope* ropeManager::GetRopeFromHandle(uint32_t handle)
+{
+	return ropeManager_getRopeFromHandle(this, handle);
+}
+
+static ropeManager** g_ropeManager;
+
+ropeManager* ropeManager::Get()
```

Comment:
GetInstance makes more sense as a name for singletons. Matches some non-inlined stuff in companion as well.